common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -193,16 +195,26 @@ def index_dictionary(self):
         """
         Return dictionary prepared with module content and type for indexing.
         """
-        result = super(CapaDescriptor, self).index_dictionary()
-        if not result:
-            result = {}
-        index = {
-            'content_type': self.INDEX_CONTENT_TYPE,
-            'problem_types': list(self.problem_types),
-            "display_name": self.display_name
+        xblock_body = super(CapaDescriptor, self).index_dictionary()
+        # Removing solution
+        capa_content = re.sub(re.compile(r"<solution>.*</solution>", re.DOTALL), "", self.data)
```

Comment:
Is it guaranteed that there will be one and only one solution per CapaModule? Will it always be surrounded by solution tags? Leaking answers to search would of course be damaging.